apps/aecore/src/aec_conductor.erl

```diff
@@ -434,16 +434,28 @@ dispatch_worker(Tag, Fun, State) ->
             State;
         false ->
             {Pid, Info} = spawn_worker(Tag, Fun),
-            State1 = block_tag(Tag, State),
+            State1 = maybe_block_tag(Tag, State),
             Workers = orddict:store(Pid, Info, State1#state.workers),
-            State1#state{workers = Workers}
+            State2 = State1#state{workers = Workers},
+            case Tag of
```

Comment:
Maybe more elegant to always return the Pid and ignore it if not needed?

Reply:
Ok... There are still some custom cleanup actions when `mining` worker is terminated/finishes... I don't like putting custom logic into workers handling, but can't come up with anything smarter to avoid that...